subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,262 @@
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="multiplePublicationsAndRepositories" title="Adding multiple publications and repositories">
+    <sourcefile file="build.gradle" snippet="publishing"/>
+</sample>
+++++
+
+The publishing plugins will create tasks that allow you to publish either of the publications to either repository. They also attach those tasks to the `publish` aggregate task. But let's say you want to restrict the binary-only publication to the external repository and the binary-with-sources publication to the internal one. To do that, you need to make the publishing _conditional_.
+
+Gradle allows you to skip any task you want based on a condition via the api:org.gradle.api.Task#onlyIf(org.gradle.api.specs.Spec)[] method. The following sample demonstrates how to implement the constraints we just mentioned:
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="publishingMavenConditionally" title="Configuring which artifacts should be published to which repositories">
+    <sourcefile file="build.gradle" snippet="task-config"/>
+    <output args="publish publishToMavenLocal"/>
+</sample>
+++++
+
+You may also want to define your own aggregate tasks to help with your workflow. For example, imagine that you have several publications that should be published to the external repository. It could be very useful to publish all of them in one go without publishing the internal ones.
+
+The following sample demonstrates how you can do this by defining an aggregate task — `publishToExternalRepository` — that depends on all the relevant publish tasks:
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="shorthandTasks" title="Defining your own shorthand tasks for publishing">
+    <sourcefile file="build.gradle" snippet="shorthand-tasks"/>
+</sample>
+++++
+
+This particular sample automatically handles the introduction or removal of the relevant publishing tasks by using api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] with the api:org.gradle.api.publish.maven.tasks.PublishToMavenRepository[] task type. You can do the same with api:org.gradle.api.publish.ivy.tasks.PublishToIvyRepository[] if you're publishing to Ivy-compatible repositories.
+
+[[sec:configuring_publishing_tasks]]
+=== Configuring publishing tasks
+
+The publishing plugins create their non-aggregate tasks after the project has been evaluated, which means you cannot directly reference them from your build script. If you would like to configure any of these tasks, you should use deferred task configuration. This can be done in a number of ways via the project's `tasks` collection.
```

Comment:
I went with this wording based on @oehme saying that we want to discourage (or simply not encourage) users to eagerly configure tasks. I also wanted to focus on _current_ behavior. I was thinking of linking to the section on deferred configuration and that feature preview, but it may just muddy the waters somewhat.

Another factor is that I don't really want to use the user manual for temporary things like a feature preview. Users are likely to miss it anyway, and it's information that inherently becomes stale quickly.